src/lib/asyncUtils.ts

```diff
@@ -0,0 +1,21 @@
+import { log } from '@/lib/telemetry';
+
+export function wrapAndLogError<T>(
+  fn: () => Promise<T>,
+  logId: string,
+  // Usually we rethrow the error. If this is true, we swallow the error and return undefined.
+  errorReturnUndefined?: boolean
```

Comment:
should we default this to true? it seems like most of the existing use cases are swallowing the errors by default

Reply:
Yes but in new react-query returning undefined causes a runtime error with no metadata, so we might as well re-throw the error that actually has metadata.